contracts/proto/resource/v1/resource.proto

```diff
@@ -63,6 +64,20 @@ message TopicResource {}
 message CollectionResource {}
 message SecretResource {}
 
+// Allow a function to define its own configuration
+message FunctionResource {
+  // name of the function resource (defaults based on file name)
+  string name = 1;
+  // The desired cpu count of the function
+  // The default will be the minimum available on the provider being deployed to
+  // and the minimum required to support requested memory
+  int32 cpus = 2;
+  // The desired memory of the function in mebibytes
+  // the default will be the minimum available on the provider
+  // being deployed to
+  int32 memory = 3;
```

Comment:
buildScripts ? @davemooreuws was asking for them..
min/max scale limits?

Reply:
@asalkeld both good suggestions, I think we should raise these as separate PRs following this one.

Reply:
Yeah, let's have that chat. Adding build scripts could help with the limitations with Prisma use right now too - i.e. we could run `npx prisma generate`